hack/etcd_cleanup.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+# Copyright 2014 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script is designed to run on master and assume etcd is running on localhost.
+# It would recursively look into etcd keys and remove any empty directory
+# It will download it's own version of etcdctl.
+# Usage: (on master the mount point may not have exec flag, use bash to run the script)
+# > bash etcd_cleanup.sh
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+TMPDIR=$(mktemp -d)
+echo "using temp directory $TMPDIR"
+MLINE=$( (MYDIR=$TMPDIR; until findmnt $MYDIR ; do MYDIR=$(dirname $MYDIR) ; done) | tail -n 1 | tr -s ' ')
```

Comment:
this is probably complicated enough to merit a comment explaining what this is finding